src/project.c

```diff
@@ -772,6 +772,40 @@ static gboolean update_config(const PropertyDialogElements *e, gboolean new_proj
 		}
 		g_free(locale_path);
 	}
+
+	gchar *project_file_dirname = g_path_get_dirname(locale_filename);
+
+	/* if the project file's directory doesn't exist, offer the user to create it */
+	if (! g_file_test(project_file_dirname, G_FILE_TEST_EXISTS))
+	{
+		gboolean create_dir = dialogs_show_question_full(NULL, GTK_STOCK_OK, GTK_STOCK_CANCEL,
+			_("Create the project's Filename directory?"),
+			_("The path \"%s\" does not exist."),
+			project_file_dirname);
+
+		/* if the project file's directory doesn't exist and the user wants to try and create it */
+		if (create_dir)
+		{
+			/* try and create the project file's directory recursively */
+			err_code = utils_mkdir(project_file_dirname, TRUE);
+			if (err_code != 0)
+			{
+				SHOW_ERR1(_("Filename path could not be created (%s)."), g_strerror(err_code));
+				gtk_widget_grab_focus(e->file_name);
+				utils_free_pointers(2, project_file_dirname, locale_filename, NULL);
+				return FALSE;
+			}
+		}
+		else
+		{
+			gtk_widget_grab_focus(e->file_name);
+			utils_free_pointers(2, project_file_dirname, locale_filename, NULL);
+			return FALSE;
+		}
```

Comment:
I haven't thought about this much (hence the TODO in my example code), are you sure this is the best thing to do?

I get that the path is tested above, but what if the directory is created in the meantime (ex. the user leaves the question dialog box up and create the directory using their file manager or the terminal)?

I could be wrong, but it seems like falling back to the old behaviour here would be ok, no?

Reply:
> I haven't thought about this much (hence the TODO in my example code), are you sure this is the best thing to do?

Yes, I think it is best this way, because it's the same behaviour as for creating `Base path`.
> 
> I get that the path is tested above, but what if the directory is created in the meantime (ex. the user leaves the question dialog box up and create the directory using their file manager or the terminal)?
> 
I just tested the existing logic for `Base path`, and if the directory is created in the meantime it doesn't recreate the directory. It works. All files inside the directory were preserved.

> I could be wrong, but it seems like falling back to the old behaviour here would be ok, no?

Reply:
> I haven't thought about this much (hence the TODO in my example code), are you sure this is the best thing to do?
> 
> I get that the path is tested above, but what if the directory is created in the meantime (ex. the user leaves the question dialog box up and create the directory using their file manager or the terminal)?
> 
> I could be wrong, but it seems like falling back to the old behaviour here would be ok, no?

But of course, if you prefer the old behaviour I make the change.

Would you like me to leave `// TODO: should this be handled?` in the `else`?

Reply:
Let's see if anyone else has any thoughts. @elextr?

Personally I would not bail out here and let it fail later if the directory wasn't created in the meantime, and maybe put a comment saying as much.

Reply:
The `utils_mkdir()` uses `g_mkdir_with_parents()` which succeeds if the directory already exists, so thats why basename behaves as it does, and should be fine to let filename work the same.